duckduckgo/Android
# Feature/request desktop site

Asana Issue URL: https://app.asana.com/0/488551667048375/529888388192278

## Description
Gives the user an option to request that the desktop site be loaded instead of a mobile-specific site. It achieves this by altering the user agent string accordingly.

Note, if the url has an obviously mobile-specific site (using the `m.xxxxxxxx.com`) format, we strip the `m.` out of the URL when jumping to the desktop version. Discussions welcomed around this.

Also included in this is the slight improvement over the general UA strings we submit; they no longer contain the device make or model in them. We do still include the rest of the details, including the Android version and `AppleWebKit/xxx.yy` version information to ensure sites render well. https://app.asana.com/0/414730916066338/537987832233112

## Steps to Test this PR:
1. Visit a site that redirects to a mobile-specific site (e.g., youtube.com)
1. Verify that you end up on the mobile site (m.youtube.com)
1. Use the overflow menu, and check to receive the desktop site
1. Verify that you end up on the desktop site
1. Repeat again to ensure you end up back on the mobile site
